pkg/provider/azure_instances.go

```diff
@@ -244,7 +245,7 @@ func (az *Cloud) InstanceShutdownByProviderID(ctx context.Context, providerID st
 		return false, err
 	}
 
-	powerStatus, err := az.VMSet.GetPowerStatusByNodeName(string(nodeName))
+	powerStatus, provisioningState, err := az.VMSet.GetPowerStatusByNodeName(string(nodeName))
```

Comment:
Can we create a new function to get provisioning state instead of piggyback the same?

Reply:
added the new interface